lib/ansible/modules/git.py

```diff
@@ -1292,6 +1294,8 @@ def main():
 
     # iface changes so need it to make decisions
     git_version_used = git_version(git_path, module)
+    if git_version_used is None:
+        module.fail_json(msg='Cannot find git executable at %s' % git_path)
```

Comment:
Enforcing git immediately does simplify things, but this codepath could have called  `exit_json` without executing git. There are two spots that have a version comparison without checking if the git_version_used is not None first, so we could add the checks at those points if we wanted to keep the original error minus unhandled traceback.

Otherwise, if the git version is enforced immediately, this can be simplified by calling `fail_json` in `git_version` (there's also a related comment that should be updated) and remove any other checks if the version is None.